packages/pigment-css-react/src/Grid.jsx

```diff
@@ -0,0 +1,311 @@
+/* eslint-disable @typescript-eslint/no-unused-expressions */
+/* eslint-disable react/jsx-filename-extension */
+import clsx from 'clsx';
+import PropTypes from 'prop-types';
+import * as React from 'react';
+
+import { generateAtomics } from './generateAtomics';
+import styled from './styled';
+
+function isGridComponent(element) {
+  // For server components `muiName` is avaialble in element.type._payload.value.muiName
+  // relevant info - https://github.com/facebook/react/blob/2807d781a08db8e9873687fccc25c0f12b4fb3d4/packages/react/src/ReactLazy.js#L45
+  // eslint-disable-next-line no-underscore-dangle
+  return element.type.muiName === 'Grid' || element.type?._payload?.value?.muiName === 'Grid';
+}
+
+export const gridAtomics = generateAtomics(({ theme }) => ({
+  conditions: Object.keys(theme.breakpoints.values).reduce((acc, breakpoint) => {
+    acc[breakpoint] = `@media (min-width: ${theme.breakpoints.values[breakpoint]}${
+      theme.breakpoints.unit ?? 'px'
+    })`;
+    return acc;
+  }, {}),
+  defaultCondition: theme.breakpoints?.keys?.[0] ?? 'xs',
+  properties: {
+    flexDirection: ['column', 'column-reverse', 'row', 'row-reverse'],
+    '--Grid-parent-column-count': ['--Grid-parent-column-count'],
+    '--Grid-parent-column-spacing': ['--Grid-parent-column-spacing'],
+    '--Grid-parent-row-spacing': ['--Grid-parent-row-spacing'],
+    '--Grid-self-column-span': ['--Grid-self-column-span'],
+    '--Grid-self-width': ['--Grid-self-width'],
+    '--Grid-self-max-width': ['--Grid-self-max-width'],
+    '--Grid-self-flex': ['--Grid-self-flex'],
+    '--Grid-self-column-spacing': ['--Grid-self-column-spacing'],
+    '--Grid-self-row-spacing': ['--Grid-self-row-spacing'],
+    '--Grid-self-offset': ['--Grid-self-offset'],
+    '--Grid-self-margin-left': ['--Grid-self-margin-left'],
```

Comment:
Yes, this is a lot of variables 😅 

But this is the only way I see we can support all the props being breakpoint aware: columns, spacing, size, offset, as well as nesting grids.

Reply:
Not a problem IMO 👍